control/runner.go

```diff
@@ -325,12 +403,17 @@ func (r *runner) HandleGomitEvent(e gomit.Event) {
 				}).Info("pool has bad key ", key)
 				continue
 			}
-
 			// attempt to find a pool whose type and name are the same, and whose version is
-			// less than newly loaded plugin.  If we find it, break out of loop.
+			// less than newly loaded plugin.
 			if core.PluginType(v.Type).String() == tnv[0] && v.Name == tnv[1] && v.Version > p.Version() {
-				pool = p
-				break
+				// See if the pool version is higher than the current highest.
+				// We only want to move subscriptions from the currentHighest
+				// because that is where subscriptions that are bound to the
+				// latest version will be.
+				if p.Version() > currentHighestVersion {
```

Comment:
On a load plugin event the only subscriptions we need to move are those that are not tied to a specific version(i.e. they will be subscribed to the latest version of a plugin). This loops through and grabs the current latest pool that has the same type/name/version so that we can move those subscriptions.